hil-test/tests/gpio_custom_handler.rs

```diff
@@ -110,4 +150,36 @@ mod tests {
         // We expect the async API to keep working even if a user handler is set.
         assert_eq!(counter, 5);
     }
+
+    #[test]
+    async fn task_that_runs_at_handlers_priority_is_not_locked_up() {
```

Comment:
Could we add a multi-core version of this test, where the drive/sense happens on each core respectively? I'd like to try and exercise the synchronization a bit more here, as if race conditions exist, they might not show up immediately :D.

Reply:
I added one but the way things have to wait for each other I'm not convinced it will actually try and race anything :)